pkg/link/spoofcheck.go

```diff
@@ -102,20 +106,31 @@ func (sc *SpoofChecker) Setup() error {
 	rulesConfig.AddRule(sc.matchMacRule(macChain.Name))
 	rulesConfig.AddRule(sc.dropRule(macChain.Name))
 
-	if err := sc.configurer.Apply(rulesConfig); err != nil {
+	rulestore, err := sc.configurer.Apply(rulesConfig)
+	if err != nil {
 		return fmt.Errorf("failed to setup spoof-check: %v", err)
 	}
+	sc.rulestore = rulestore
 
 	return nil
 }
 
+func (sc *SpoofChecker) StoredRules() (*nft.Config, error) {
+	if sc.rulestore != nil {
+		rulestore := sc.rulestore
+		sc.rulestore = nil
+		return rulestore, nil
+	}
+	return sc.configurer.Read(listChainBridgeNatPrerouting()...)
+}
+
 // Teardown removes the interface and mac-address specific chains and their rules.
 // The table and base-chain are expected to survive while the base-chain rule that matches the
 // interface is removed.
 func (sc *SpoofChecker) Teardown() error {
 	ifaceChain := sc.ifaceChain()
-	currentConfig, ifaceMatchRuleErr := sc.configurer.Read(listChainBridgeNatPrerouting()...)
-	if ifaceMatchRuleErr == nil {
+	currentConfig, ifaceMatchRuleErr := sc.StoredRules()
+	if currentConfig != nil {
```

Comment:
In Go, it is common to assume the return value is undefined when an error returns.
It will be better to keep this here as well, i.e. check against the error and only then proceed to check against the value.